sklearn/tree/_classes.py

```diff
@@ -1254,7 +1392,14 @@ def __init__(
             ccp_alpha=ccp_alpha,
         )
 
-    def fit(self, X, y, sample_weight=None, check_input=True):
+    def fit(
+        self,
+        X,
+        y,
+        sample_weight=None,
+        check_input=True,
+        classes=None,
```

Comment:
Do we need to add `classes` keyword argument, or is this due to an outdated API?

Reply:
This is to accommodate the tree building. Not sklearn api I think. The tree needs to know all the potential classes from the beginning, as not all classes are necessarily included in the first batch of data. If new classes are added later there's errors on y dimensions.